pkg/config/config.go

```diff
@@ -149,8 +149,8 @@ func (c *Config) Render(properties map[string]interface{}) (string, error) {
 		return "", nil
 	}
 
-	templatePath := c.Template.Name()
-	if t, ok := c.Template.(template.FileBasedTemplate); ok {
+	templatePath := c.Template.Id()
+	if t, ok := c.Template.(*template.FileBasedTemplate); ok {
 		templatePath = t.FilePath()
 	}
```

Comment:
i guess you can delete these lines

Reply:
Upon closer inspection, no this is still needed. 

I didn't make Filepath() part of the interface and while both implementations currently have a Filepath() they behave differently. 
FileBasedTemplate always has a path and has a `Filepath() string` signature.
InMemoryTemplate _might_ have a defined path and has a `Filepath() *string` signature.

Usage is a bit odd at the moment... outside of tests, Render() will always get a FileBasedTemplate right now. 
The config writer might get InMemoryTemplates from download (no path), or convert (fixed path), and I think(!) will currently never get a FileBasedTemplate, but handles all in: pkg/persistence/config/writer/config_writer.go:611

I'm not sure of a better way to handle this at the moment. Pulling the optional Filepath *string method into the interface feels odd as well. 
If you have a good refactoring idea, let me know! Else maybe we leave it as is until someone has an idea to refactor?

Reply:
With this changes `ID()` method for `FileBasedTemplate` returns path, but for `InMemory` real ID.  This addresses my complaint for different behavior for `ID()`

Reply:
Technically neither the Id or Name make much sense here. 
The filepath is only ever included as a hint in errors that happend during rendering.
Additionally at least right now a non FileBasedTemplate will never be rendered, as that happens only during deploy. 

I've removed the unecessary and confusing initialization with the ID and just leave the path variable empyt in the unlikely case that a template without a path is being rendered.